<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="generator" content="Source Themes Academic 3.3.0">
  <meta name="generator" content="Hugo 0.53" />

  

  
  
  
  
  
    
    
    
  
  

  <meta name="author" content="Erco 1976">

  
  
  
    
  
  <meta name="description" content=" Ribosomal RNA (rRNA) genes are the most widely used data source in phylogenetic reconstruction. They are highly structured, with large parts of the mol- ecules exhibiting very strong conservation of their base pairing patterns. Since also sequence conservation varies dramatically between different regions of rRNA genes, these data are informative on a wide range of phylogenetic timescales, ranging from recent to ancient splits. But individual columns of rRNA alignments are not inde- pendent. This is a consequence of biologically functional secondary structures with highly varying degrees of conservation. Therefore, any correlations within sequence alignments of structurally functional RNA will distort the phylogenetic signal and/or lead to gross overestimates of tree stability. On the other hand, alignment accuracy can be improved substantially by incorporating secondary structure conservation. Maximum Likelihood and Bayesian approaches are amenable to using RNA-specific substitution models that treat conserved base pairs appropriately, but they require accurate secondary structure models as input. Structure prediction algorithms for single RNA sequences are well known and widely used; but it is not straightforward to apply these predictions for phylogenetic purposes. The main limitation is that accuracy of thermodynamic folding algorithms declines sharply as the length of the RNA increases, this is in part due to inaccuracies within the thermodynamic folding parameters, in part caused by the kinetics of folding process and tertiary interac- tions, and even more in the fact that RNA and protein components of the ribosome are tightly packed and thus mutually influence their folds. The functional rRNA struc- tures, therefore, cannot reasonably be expected to be identical with the minimum free energy structures of isolated rRNAs computed by thermodynamic folding algorithms. Our RNAsalsa approach has been the first accurate and easy-to-use tool to address those implications mentioned. It is available for computing structure-aware align- ments and consensus structures and can deal with large sets of large rRNAs. Multiple pairwise sequence alignments, intersections with individualized structural constraint modeling, and computationally derived collections of optimized subgroup consen- sus structures allow to extract a refined individualized structure constraint for each single sequence by means of a majority voting procedure. That way an extensively optimized set of individual constraints serves in minimum energy folding to obtain the final structural annotation for each input sequence. Further, the structure-anno- tated input sequences are aligned by a structure-aware alignment method, resulting in a final alignment and a corresponding consensus structure which are ready for phylogenetic analysis based on RNA-specific substitution models. The difference to DNA substitution models is that structure-aware models consider pairs of nucleotides as their elementary states rather than single nucleotides. Further, studies suggest that RNA secondary structures are by themselves also phylogenetically informative since structural variations are specific to individual clades. This has been reported for a wide variety of non-coding RNAs, including rRNAs, 7SK RNAs, and mitochondrial tRNAs. But in all cases the massive epistasis needs to be taken into account explicitly which is introduced by strong selection on secondary structure features.">

  
  <link rel="alternate" hreflang="en-us" href="https://htafer.github.io/publication/tafer2014rnaphylogeny/">

  


  

  

  

  

  

  

  
  
  
  <meta name="theme-color" content="#2962ff">
  

  
  
  
  
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.1.3/css/bootstrap.min.css" integrity="sha256-eSi1q2PG6J7g7ib17yAaWMcrr5GrtohYChqibrV7PBE=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css" integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.0/css/all.css" integrity="sha384-aOkxzJ5uQz7WBObEZcHvV5JvRW3TUc2rNPA7pe3AwnsUohiw1Vj2Rgx2KSOkF5+h" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.css" integrity="sha256-ygkqlh3CYSUri3LhQxzdcm0n1EQvH2Y+U5S2idbLtxs=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-mfizz/2.4.1/font-mfizz.min.css" integrity="sha512-Cdvnk1SFWqcb3An6gMyqDRH40Js8qmsWcSK10I2gSifCe2LilaPMsHd6DldEvQ3uIlCb1qdRUrNeAFFleOu4xQ==" crossorigin="anonymous">

    
    
    
      
    
    
      
      
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/sunburst.min.css" crossorigin="anonymous">
      
    

    

    

  

  
  
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Montserrat:400,700|Roboto:400,400italic,700|Roboto+Mono|Major+Mono+Display|VT323">
  

  <link rel="stylesheet" href="/styles.css">
  

  
  
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-132338789-1', 'auto');
      ga('set', 'anonymizeIp', true);
      ga('require', 'eventTracker');
      ga('require', 'outboundLinkTracker');
      ga('require', 'urlChangeTracker');
      ga('send', 'pageview');
    </script>
    <script async src="//www.google-analytics.com/analytics.js"></script>
    
    <script async src="https://cdnjs.cloudflare.com/ajax/libs/autotrack/2.4.1/autotrack.js" integrity="sha512-HUmooslVKj4m6OBu0OgzjXXr+QuFYy/k7eLI5jdeEy/F4RSgMn6XRWRGkFi5IFaFgy7uFTkegp3Z0XnJf3Jq+g==" crossorigin="anonymous"></script>
    
  
  

  
  <link rel="alternate" href="https://htafer.github.io/index.xml" type="application/rss+xml" title="Code Blocks">
  <link rel="feed" href="https://htafer.github.io/index.xml" type="application/rss+xml" title="Code Blocks">
  

  <link rel="manifest" href="/site.webmanifest">
  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/icon-192.png">

  <link rel="canonical" href="https://htafer.github.io/publication/tafer2014rnaphylogeny/">

  
  
  
  
    
    
  
  <meta property="twitter:card" content="summary">
  
  <meta property="og:site_name" content="Code Blocks">
  <meta property="og:url" content="https://htafer.github.io/publication/tafer2014rnaphylogeny/">
  <meta property="og:title" content="RNA in phylogenetic reconstruction | Code Blocks">
  <meta property="og:description" content=" Ribosomal RNA (rRNA) genes are the most widely used data source in phylogenetic reconstruction. They are highly structured, with large parts of the mol- ecules exhibiting very strong conservation of their base pairing patterns. Since also sequence conservation varies dramatically between different regions of rRNA genes, these data are informative on a wide range of phylogenetic timescales, ranging from recent to ancient splits. But individual columns of rRNA alignments are not inde- pendent. This is a consequence of biologically functional secondary structures with highly varying degrees of conservation. Therefore, any correlations within sequence alignments of structurally functional RNA will distort the phylogenetic signal and/or lead to gross overestimates of tree stability. On the other hand, alignment accuracy can be improved substantially by incorporating secondary structure conservation. Maximum Likelihood and Bayesian approaches are amenable to using RNA-specific substitution models that treat conserved base pairs appropriately, but they require accurate secondary structure models as input. Structure prediction algorithms for single RNA sequences are well known and widely used; but it is not straightforward to apply these predictions for phylogenetic purposes. The main limitation is that accuracy of thermodynamic folding algorithms declines sharply as the length of the RNA increases, this is in part due to inaccuracies within the thermodynamic folding parameters, in part caused by the kinetics of folding process and tertiary interac- tions, and even more in the fact that RNA and protein components of the ribosome are tightly packed and thus mutually influence their folds. The functional rRNA struc- tures, therefore, cannot reasonably be expected to be identical with the minimum free energy structures of isolated rRNAs computed by thermodynamic folding algorithms. Our RNAsalsa approach has been the first accurate and easy-to-use tool to address those implications mentioned. It is available for computing structure-aware align- ments and consensus structures and can deal with large sets of large rRNAs. Multiple pairwise sequence alignments, intersections with individualized structural constraint modeling, and computationally derived collections of optimized subgroup consen- sus structures allow to extract a refined individualized structure constraint for each single sequence by means of a majority voting procedure. That way an extensively optimized set of individual constraints serves in minimum energy folding to obtain the final structural annotation for each input sequence. Further, the structure-anno- tated input sequences are aligned by a structure-aware alignment method, resulting in a final alignment and a corresponding consensus structure which are ready for phylogenetic analysis based on RNA-specific substitution models. The difference to DNA substitution models is that structure-aware models consider pairs of nucleotides as their elementary states rather than single nucleotides. Further, studies suggest that RNA secondary structures are by themselves also phylogenetically informative since structural variations are specific to individual clades. This has been reported for a wide variety of non-coding RNAs, including rRNAs, 7SK RNAs, and mitochondrial tRNAs. But in all cases the massive epistasis needs to be taken into account explicitly which is introduced by strong selection on secondary structure features."><meta property="og:image" content="https://htafer.github.io/img/icon-192.png">
  <meta property="og:locale" content="en-us">
  
  <meta property="article:published_time" content="2014-12-31T00:00:00&#43;00:00">
  
  <meta property="article:modified_time" content="2014-12-31T00:00:00&#43;00:00">
  

  

<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/cookieconsent2/3.0.3/cookieconsent.min.css">
<script src="//cdnjs.cloudflare.com/ajax/libs/cookieconsent2/3.0.3/cookieconsent.min.js"></script>
<script>
  window.addEventListener("load", function(){
    window.cookieconsent.initialise({
      "palette": {
        "popup": {
          "background": "#2962ff",
          "text": "#fff"
        },
        "button": {
          "background": "#fff",
          "text": "#2962ff"
        }
      },
      "theme": "classic",
      "content": {
        "message": "This website uses cookies to ensure you get the best experience on our website.",
        "dismiss": "Got it!",
        "link": "Learn more",
        "href": "https://cookies.insites.com"
      }
    })});
</script>


  

  <title>RNA in phylogenetic reconstruction | Code Blocks</title>

</head>
<body id="top" data-spy="scroll" data-target="#TableOfContents" data-offset="71" >
  <aside class="search-results" id="search">
  <div class="container">
    <section class="search-header">

      <div class="row no-gutters justify-content-between mb-3">
        <div class="col-6">
          <h1>Search</h1>
        </div>
        <div class="col-6 col-search-close">
          <a class="js-search" href="#"><i class="fas fa-times-circle text-muted" aria-hidden="true"></i></a>
        </div>
      </div>

      <div id="search-box">
        
        <input name="q" id="search-query" placeholder="Search..." autocapitalize="off"
        autocomplete="off" autocorrect="off" role="textbox" spellcheck="false" type="search">
        
      </div>

    </section>
    <section class="section-search-results">

      <div id="search-hits">
        
      </div>

    </section>
  </div>
</aside>


<nav class="navbar navbar-light fixed-top navbar-expand-lg py-0" id="navbar-main">
  <div class="container">

    
      <a class="navbar-brand" href="/">Code Blocks</a>
      
      <button type="button" class="navbar-toggler" data-toggle="collapse"
              data-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
        <span><i class="fas fa-bars"></i></span>
      </button>
      

    
    <div class="collapse navbar-collapse" id="navbar">

      
      
      <ul class="navbar-nav mr-auto">
        

        

        
        
        
          
        

        <li class="nav-item">
          <a class="nav-link" href="/#about">
            
            <span>Home</span>
            
          </a>
        </li>

        
        

        

        
        
        
          
        

        <li class="nav-item">
          <a class="nav-link" href="/post">
            
            <span>Posts</span>
            
          </a>
        </li>

        
        

        

        
        
        
          
        

        <li class="nav-item">
          <a class="nav-link" href="/#posts">
            
            <span>Recent Posts</span>
            
          </a>
        </li>

        
        

        

        
        
        
          
        

        <li class="nav-item">
          <a class="nav-link" href="/#projects">
            
            <span>Projects</span>
            
          </a>
        </li>

        
        

        

        
        
        
          
        

        <li class="nav-item">
          <a class="nav-link" href="/publication">
            
            <span>Publications</span>
            
          </a>
        </li>

        
        

        

        
        
        
          
        

        <li class="nav-item">
          <a class="nav-link" href="/#publications">
            
            <span>Recent Publications</span>
            
          </a>
        </li>

        
        

      
      </ul>
      <ul class="navbar-nav ml-auto">
      

        

        
        <li class="nav-item">
          <a class="nav-link js-search" href="#"><i class="fas fa-search" aria-hidden="true"></i></a>
        </li>
        

        

        
        <li class="nav-item">
          <a class="nav-link js-dark-toggle" href="#"><i class="fas fa-moon" aria-hidden="true"></i></a>
        </li>
        

      </ul>

    </div>
  </div>
</nav>

<div class="pub" itemscope itemtype="http://schema.org/CreativeWork">

  












  

  
  
  
<div class="article-container pt-3">
  <h1 itemprop="name">RNA in phylogenetic reconstruction</h1>

  

  
    



<meta content="2014-12-31 00:00:00 &#43;0000 UTC" itemprop="datePublished">
<meta content="2014-12-31 00:00:00 &#43;0000 UTC" itemprop="dateModified">

<div class="article-metadata">

  
  
  
  
  <div>
    



  <span itemscope itemprop="author" itemtype="http://schema.org/Person">
      <span itemprop="name">
        

      
      
      <a href="/authors/roman-r.-stocsits/">Roman R. Stocsits</a></span></span>, 
  <span itemscope itemprop="author" itemtype="http://schema.org/Person">
      <span itemprop="name">
        

      
      
      <a href="/authors/harald-letsch/">Harald Letsch</a></span></span>, 
  <span itemscope itemprop="author" itemtype="http://schema.org/Person">
      <span itemprop="name">
        

      
      
      <a href="/authors/karen-meusemann/">Karen Meusemann</a></span></span>, 
  <span itemscope itemprop="author" itemtype="http://schema.org/Person">
      <span itemprop="name">
        

      
      
      <a href="/authors/bjoern-m.-reumont/">Bjoern M. Reumont</a></span></span>, 
  <span itemscope itemprop="author" itemtype="http://schema.org/Person">
      <span itemprop="name">
        

      
      
      <a href="/authors/bernhard-misof/">Bernhard Misof</a></span></span>, 
  <span itemscope itemprop="author" itemtype="http://schema.org/Person">
      <span itemprop="name">
        

      
      
      <a href="/authors/jana-hertel/">Jana Hertel</a></span></span>, 
  <span itemscope itemprop="author" itemtype="http://schema.org/Person">
      <span itemprop="name">
        

      
      
      <a href="/authors/hakim-tafer/">Hakim Tafer</a></span></span>, 
  <span itemscope itemprop="author" itemtype="http://schema.org/Person">
      <span itemprop="name">
        

      
      
      <a href="/authors/peter-f.-stadler/">Peter F. Stadler</a></span></span>
  



  </div>
  
  

  
  <span class="article-date">
    
    
      
    
    <time>December 2014</time>
  </span>
  

  

  

  
  
  <span class="middot-divider"></span>
  <a href="/publication/tafer2014rnaphylogeny/#disqus_thread"></a>
  

  

  
    
<div class="share-box" aria-hidden="true">
  <ul class="share">
    <li>
      <a class="twitter"
         href="https://twitter.com/intent/tweet?text=RNA%20in%20phylogenetic%20reconstruction&amp;url=https%3a%2f%2fhtafer.github.io%2fpublication%2ftafer2014rnaphylogeny%2f"
         target="_blank" rel="noopener">
        <i class="fab fa-twitter"></i>
      </a>
    </li>
    <li>
      <a class="facebook"
         href="https://www.facebook.com/sharer.php?u=https%3a%2f%2fhtafer.github.io%2fpublication%2ftafer2014rnaphylogeny%2f"
         target="_blank" rel="noopener">
        <i class="fab fa-facebook-f"></i>
      </a>
    </li>
    <li>
      <a class="linkedin"
         href="https://www.linkedin.com/shareArticle?mini=true&amp;url=https%3a%2f%2fhtafer.github.io%2fpublication%2ftafer2014rnaphylogeny%2f&amp;title=RNA%20in%20phylogenetic%20reconstruction"
         target="_blank" rel="noopener">
        <i class="fab fa-linkedin-in"></i>
      </a>
    </li>
    <li>
      <a class="weibo"
         href="http://service.weibo.com/share/share.php?url=https%3a%2f%2fhtafer.github.io%2fpublication%2ftafer2014rnaphylogeny%2f&amp;title=RNA%20in%20phylogenetic%20reconstruction"
         target="_blank" rel="noopener">
        <i class="fab fa-weibo"></i>
      </a>
    </li>
    <li>
      <a class="email"
         href="mailto:?subject=RNA%20in%20phylogenetic%20reconstruction&amp;body=https%3a%2f%2fhtafer.github.io%2fpublication%2ftafer2014rnaphylogeny%2f">
        <i class="fas fa-envelope"></i>
      </a>
    </li>
  </ul>
</div>


  

</div>

    







  









  
</div>



  <div class="article-container">

    
    <h3>Abstract</h3>
    <p class="pub-abstract" itemprop="text">Ribosomal RNA (rRNA) genes are the most widely used data source in phylogenetic reconstruction. They are highly structured, with large parts of the mol- ecules exhibiting very strong conservation of their base pairing patterns. Since also sequence conservation varies dramatically between different regions of rRNA genes, these data are informative on a wide range of phylogenetic timescales, ranging from recent to ancient splits. But individual columns of rRNA alignments are not inde- pendent. This is a consequence of biologically functional secondary structures with highly varying degrees of conservation. Therefore, any correlations within sequence alignments of structurally functional RNA will distort the phylogenetic signal and/or lead to gross overestimates of tree stability. On the other hand, alignment accuracy can be improved substantially by incorporating secondary structure conservation. Maximum Likelihood and Bayesian approaches are amenable to using RNA-specific substitution models that treat conserved base pairs appropriately, but they require accurate secondary structure models as input. Structure prediction algorithms for single RNA sequences are well known and widely used; but it is not straightforward to apply these predictions for phylogenetic purposes. The main limitation is that accuracy of thermodynamic folding algorithms declines sharply as the length of the RNA increases, this is in part due to inaccuracies within the thermodynamic folding parameters, in part caused by the kinetics of folding process and tertiary interac- tions, and even more in the fact that RNA and protein components of the ribosome are tightly packed and thus mutually influence their folds. The functional rRNA struc- tures, therefore, cannot reasonably be expected to be identical with the minimum free energy structures of isolated rRNAs computed by thermodynamic folding algorithms. Our RNAsalsa approach has been the first accurate and easy-to-use tool to address those implications mentioned. It is available for computing structure-aware align- ments and consensus structures and can deal with large sets of large rRNAs. Multiple pairwise sequence alignments, intersections with individualized structural constraint modeling, and computationally derived collections of optimized subgroup consen- sus structures allow to extract a refined individualized structure constraint for each single sequence by means of a majority voting procedure. That way an extensively optimized set of individual constraints serves in minimum energy folding to obtain the final structural annotation for each input sequence. Further, the structure-anno- tated input sequences are aligned by a structure-aware alignment method, resulting in a final alignment and a corresponding consensus structure which are ready for phylogenetic analysis based on RNA-specific substitution models. The difference to DNA substitution models is that structure-aware models consider pairs of nucleotides as their elementary states rather than single nucleotides. Further, studies suggest that RNA secondary structures are by themselves also phylogenetically informative since structural variations are specific to individual clades. This has been reported for a wide variety of non-coding RNAs, including rRNAs, 7SK RNAs, and mitochondrial tRNAs. But in all cases the massive epistasis needs to be taken into account explicitly which is introduced by strong selection on secondary structure features.</p>
    

    
    <div class="row">
      <div class="col-md-1"></div>
      <div class="col-md-10">
        <div class="row">
          <div class="col-12 col-md-3 pub-row-heading">Type</div>
          <div class="col-12 col-md-9">
            

            
            
            <a href="/publication/#1">
              Conference paper
            </a>
            
          </div>
        </div>
      </div>
      <div class="col-md-1"></div>
    </div>
    <div class="d-md-none space-below"></div>
    

    

    <div class="space-below"></div>

    <div class="article-style"></div>

    




    






  
  
    
  
  








  </div>
</div>



<div class="container">
  <footer class="site-footer">
  

  <p class="powered-by">
    &copy; 2018-2019 &middot; 

    Powered by the
    <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
    <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

    
    <span class="float-right" aria-hidden="true">
      <a href="#" id="back_to_top">
        <span class="button_icon">
          <i class="fas fa-chevron-up fa-2x"></i>
        </span>
      </a>
    </span>
    
  </p>
</footer>

</div>


<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title">Cite</h5>
        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body">
        <pre><code class="tex hljs"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-outline-primary my-1 js-copy-cite" href="#" target="_blank">
          <i class="fas fa-copy"></i> Copy
        </a>
        <a class="btn btn-outline-primary my-1 js-download-cite" href="#" target="_blank">
          <i class="fas fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

    

    
    

    
    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha512-+NqPlbbtM1QqiK8ZAo4Yrj2c4lNQoGv8P79DPtKzj++l5jnN39rHA/xsqn8zE9l0uSoxaCdrOgFs6yjyfbBxSg==" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.3/imagesloaded.pkgd.min.js" integrity="sha512-umsR78NN0D23AzgoZ11K7raBD+R6hqKojyBZs1w8WvYlsI+QuKRGBx3LFCwhatzBunCjDuJpDHwxD13sLMbpRA==" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.1.3/js/bootstrap.min.js" integrity="sha256-VsEqElsCHSGmnmHXGQzvoWjWwoznFSZc6hs7ARLRacQ=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.4/isotope.pkgd.min.js" integrity="sha512-VDBOIlDbuC4VWxGJNmuFRQ0Li0SKkDpmGyuhAG5LTDLd/dJ/S0WMVxriR2Y+CyPL5gzjpN4f/6iqWVBJlht0tQ==" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.js" integrity="sha256-X5PoE3KU5l+JcX+w09p/wHl9AzK333C4hJ2I9S5mD4M=" crossorigin="anonymous"></script>

      
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js" integrity="sha256-/BfiIkHlHoVihZdc6TFuj7MmJ0TWcWsMXkeDFwhi0zw=" crossorigin="anonymous"></script>
        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/r.min.js"></script>
        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/lisp.min.js"></script>
        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/bash.min.js"></script>
        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/C.min.js"></script>
        
      

      
      
    

    
    

    
    
    
    <script id="dsq-count-scr" src="//Erco1976.disqus.com/count.js" async></script>
    

    
    
    <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script>
      const search_index_filename = "/index.json";
      const i18n = {
        'placeholder': "Search...",
        'results': "results found",
        'no_results': "No results found"
      };
      const content_type = {
        'post': "Posts",
        'project': "Projects",
        'publication' : "Publications",
        'talk' : "Talks"
        };
    </script>
    

    
    

    
    
    <script id="search-hit-fuse-template" type="text/x-template">
      <div class="search-hit" id="summary-{{key}}">
      <div class="search-hit-content">
        <div class="search-hit-name">
          <a href="{{relpermalink}}">{{title}}</a>
          <div class="article-metadata search-hit-type">{{type}}</div>
          <p class="search-hit-description">{{snippet}}</p>
        </div>
      </div>
      </div>
    </script>
    

    
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/3.2.1/fuse.min.js" integrity="sha256-VzgmKYmhsGNNN4Ph1kMW+BjoYJM2jV5i4IlFoeZA9XI=" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>
    

    
    

    
    
    
    
    
    
    
      
    
    
    
    
    <script src="/js/academic.min.7903dc64949b0cecbfa70e940e162fa7.js"></script>

  </body>
</html>

