<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="generator" content="Source Themes Academic 3.3.0">
  <meta name="generator" content="Hugo 0.53" />

  

  
  
  
  
  
    
    
    
  
  

  <meta name="author" content="Erco 1976">

  
  
  
    
  
  <meta name="description" content="I have been terminating a project where proteomic profiles of of a plant seed have been done at a various developmental stages. On of the question was to look at when which enzyme shows up in KEGG pathway.
DATA In terms of data we have for each stage a list of proteins with at least one protein-specific peptide. My main work was to look at this list of gene and find enriched functional annotation terms.">

  
  <link rel="alternate" hreflang="en-us" href="https://htafer.github.io/post/kegg2/">

  


  

  

  

  

  

  

  
  
  
  <meta name="theme-color" content="#2962ff">
  

  
  
  
  
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.1.3/css/bootstrap.min.css" integrity="sha256-eSi1q2PG6J7g7ib17yAaWMcrr5GrtohYChqibrV7PBE=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css" integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.0/css/all.css" integrity="sha384-aOkxzJ5uQz7WBObEZcHvV5JvRW3TUc2rNPA7pe3AwnsUohiw1Vj2Rgx2KSOkF5+h" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.css" integrity="sha256-ygkqlh3CYSUri3LhQxzdcm0n1EQvH2Y+U5S2idbLtxs=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-mfizz/2.4.1/font-mfizz.min.css" integrity="sha512-Cdvnk1SFWqcb3An6gMyqDRH40Js8qmsWcSK10I2gSifCe2LilaPMsHd6DldEvQ3uIlCb1qdRUrNeAFFleOu4xQ==" crossorigin="anonymous">

    
    
    
      
    
    
      
      
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/sunburst.min.css" crossorigin="anonymous">
      
    

    

    

  

  
  
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Montserrat:400,700|Roboto:400,400italic,700|Roboto+Mono|Major+Mono+Display|VT323">
  

  <link rel="stylesheet" href="/styles.css">
  

  
  
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-132338789-1', 'auto');
      ga('set', 'anonymizeIp', true);
      ga('require', 'eventTracker');
      ga('require', 'outboundLinkTracker');
      ga('require', 'urlChangeTracker');
      ga('send', 'pageview');
    </script>
    <script async src="//www.google-analytics.com/analytics.js"></script>
    
    <script async src="https://cdnjs.cloudflare.com/ajax/libs/autotrack/2.4.1/autotrack.js" integrity="sha512-HUmooslVKj4m6OBu0OgzjXXr+QuFYy/k7eLI5jdeEy/F4RSgMn6XRWRGkFi5IFaFgy7uFTkegp3Z0XnJf3Jq+g==" crossorigin="anonymous"></script>
    
  
  

  
  <link rel="alternate" href="https://htafer.github.io/index.xml" type="application/rss+xml" title="Code Blocks">
  <link rel="feed" href="https://htafer.github.io/index.xml" type="application/rss+xml" title="Code Blocks">
  

  <link rel="manifest" href="/site.webmanifest">
  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/icon-192.png">

  <link rel="canonical" href="https://htafer.github.io/post/kegg2/">

  
  
  
  
    
    
  
  <meta property="twitter:card" content="summary">
  
  <meta property="og:site_name" content="Code Blocks">
  <meta property="og:url" content="https://htafer.github.io/post/kegg2/">
  <meta property="og:title" content="KEGG analysis of regulated genes | Code Blocks">
  <meta property="og:description" content="I have been terminating a project where proteomic profiles of of a plant seed have been done at a various developmental stages. On of the question was to look at when which enzyme shows up in KEGG pathway.
DATA In terms of data we have for each stage a list of proteins with at least one protein-specific peptide. My main work was to look at this list of gene and find enriched functional annotation terms."><meta property="og:image" content="https://htafer.github.io/img/icon-192.png">
  <meta property="og:locale" content="en-us">
  
  <meta property="article:published_time" content="2019-02-08T00:00:00&#43;01:00">
  
  <meta property="article:modified_time" content="2019-02-08T00:00:00&#43;01:00">
  

  

<link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/cookieconsent2/3.0.3/cookieconsent.min.css">
<script src="//cdnjs.cloudflare.com/ajax/libs/cookieconsent2/3.0.3/cookieconsent.min.js"></script>
<script>
  window.addEventListener("load", function(){
    window.cookieconsent.initialise({
      "palette": {
        "popup": {
          "background": "#2962ff",
          "text": "#fff"
        },
        "button": {
          "background": "#fff",
          "text": "#2962ff"
        }
      },
      "theme": "classic",
      "content": {
        "message": "This website uses cookies to ensure you get the best experience on our website.",
        "dismiss": "Got it!",
        "link": "Learn more",
        "href": "https://cookies.insites.com"
      }
    })});
</script>


  

  <title>KEGG analysis of regulated genes | Code Blocks</title>

</head>
<body id="top" data-spy="scroll" data-target="#TableOfContents" data-offset="71" >
  <aside class="search-results" id="search">
  <div class="container">
    <section class="search-header">

      <div class="row no-gutters justify-content-between mb-3">
        <div class="col-6">
          <h1>Search</h1>
        </div>
        <div class="col-6 col-search-close">
          <a class="js-search" href="#"><i class="fas fa-times-circle text-muted" aria-hidden="true"></i></a>
        </div>
      </div>

      <div id="search-box">
        
        <input name="q" id="search-query" placeholder="Search..." autocapitalize="off"
        autocomplete="off" autocorrect="off" role="textbox" spellcheck="false" type="search">
        
      </div>

    </section>
    <section class="section-search-results">

      <div id="search-hits">
        
      </div>

    </section>
  </div>
</aside>


<nav class="navbar navbar-light fixed-top navbar-expand-lg py-0" id="navbar-main">
  <div class="container">

    
      <a class="navbar-brand" href="/">Code Blocks</a>
      
      <button type="button" class="navbar-toggler" data-toggle="collapse"
              data-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
        <span><i class="fas fa-bars"></i></span>
      </button>
      

    
    <div class="collapse navbar-collapse" id="navbar">

      
      
      <ul class="navbar-nav mr-auto">
        

        

        
        
        
          
        

        <li class="nav-item">
          <a class="nav-link" href="/#about">
            
            <span>Home</span>
            
          </a>
        </li>

        
        

        

        
        
        
          
        

        <li class="nav-item">
          <a class="nav-link" href="/post">
            
            <span>Posts</span>
            
          </a>
        </li>

        
        

        

        
        
        
          
        

        <li class="nav-item">
          <a class="nav-link" href="/#posts">
            
            <span>Recent Posts</span>
            
          </a>
        </li>

        
        

        

        
        
        
          
        

        <li class="nav-item">
          <a class="nav-link" href="/#projects">
            
            <span>Projects</span>
            
          </a>
        </li>

        
        

        

        
        
        
          
        

        <li class="nav-item">
          <a class="nav-link" href="/publication">
            
            <span>Publications</span>
            
          </a>
        </li>

        
        

        

        
        
        
          
        

        <li class="nav-item">
          <a class="nav-link" href="/#publications">
            
            <span>Recent Publications</span>
            
          </a>
        </li>

        
        

      
      </ul>
      <ul class="navbar-nav ml-auto">
      

        

        
        <li class="nav-item">
          <a class="nav-link js-search" href="#"><i class="fas fa-search" aria-hidden="true"></i></a>
        </li>
        

        

        
        <li class="nav-item">
          <a class="nav-link js-dark-toggle" href="#"><i class="fas fa-moon" aria-hidden="true"></i></a>
        </li>
        

      </ul>

    </div>
  </div>
</nav>


<article class="article" itemscope itemtype="http://schema.org/Article">

  












  

  
  
  
<div class="article-container pt-3">
  <h1 itemprop="name">KEGG analysis of regulated genes</h1>

  

  
    



<meta content="2019-02-08 00:00:00 &#43;0100 CET" itemprop="datePublished">
<meta content="2019-02-08 00:00:00 &#43;0100 CET" itemprop="dateModified">

<div class="article-metadata">

  
  

  
  <span class="article-date">
    
    
      
    
    <time>Feb 8, 2019</time>
  </span>
  

  

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    5 min read
  </span>
  

  
  
  <span class="middot-divider"></span>
  <a href="/post/kegg2/#disqus_thread"></a>
  

  

  
    
<div class="share-box" aria-hidden="true">
  <ul class="share">
    <li>
      <a class="twitter"
         href="https://twitter.com/intent/tweet?text=KEGG%20analysis%20of%20regulated%20genes&amp;url=https%3a%2f%2fhtafer.github.io%2fpost%2fkegg2%2f"
         target="_blank" rel="noopener">
        <i class="fab fa-twitter"></i>
      </a>
    </li>
    <li>
      <a class="facebook"
         href="https://www.facebook.com/sharer.php?u=https%3a%2f%2fhtafer.github.io%2fpost%2fkegg2%2f"
         target="_blank" rel="noopener">
        <i class="fab fa-facebook-f"></i>
      </a>
    </li>
    <li>
      <a class="linkedin"
         href="https://www.linkedin.com/shareArticle?mini=true&amp;url=https%3a%2f%2fhtafer.github.io%2fpost%2fkegg2%2f&amp;title=KEGG%20analysis%20of%20regulated%20genes"
         target="_blank" rel="noopener">
        <i class="fab fa-linkedin-in"></i>
      </a>
    </li>
    <li>
      <a class="weibo"
         href="http://service.weibo.com/share/share.php?url=https%3a%2f%2fhtafer.github.io%2fpost%2fkegg2%2f&amp;title=KEGG%20analysis%20of%20regulated%20genes"
         target="_blank" rel="noopener">
        <i class="fab fa-weibo"></i>
      </a>
    </li>
    <li>
      <a class="email"
         href="mailto:?subject=KEGG%20analysis%20of%20regulated%20genes&amp;body=https%3a%2f%2fhtafer.github.io%2fpost%2fkegg2%2f">
        <i class="fas fa-envelope"></i>
      </a>
    </li>
  </ul>
</div>


  

</div>

    















  
</div>



  <div class="article-container">

    <div class="article-style" itemprop="articleBody">
      

<p>I have been terminating a project where proteomic profiles of
of a plant seed have been done at a various developmental
stages. On of the question was to look at <strong>when which enzyme shows up
in KEGG pathway.</strong></p>

<h2 id="data">DATA</h2>

<p>In terms of data we have for each stage a list of proteins
with at least one protein-specific peptide. My main work was to look
at this list of gene and find enriched functional annotation
terms. The annotation was done with Interproscan, eggNog, Cazy, Merops
and TCDB. Gene Ontology and KEGG pathways were extracted from <a href="http://eggnogdb.embl.de/" target="_blank">eggNog</a>
and <a href="https://www.ebi.ac.uk/interpro/search/sequence-search" target="_blank">Interproscan</a>. The enrichment was made in R with the package
GOstats</p>

<p><a id="table--Table summary enrichment"></a>
<div class="table-caption">
  <span class="table-number"><a href="#table--Table summary enrichment">Table 1</a></span>:
  Overview of the number of proteins with at least 1 peptide in each developmental stage.
</div></p>

<table>
<thead>
<tr>
<th></th>
<th>GW</th>
<th>GH</th>
<th>GB</th>
<th>YB</th>
<th>BB</th>
<th>DRY</th>
</tr>
</thead>

<tbody>
<tr>
<td>Proteins with at least 1 peptide</td>
<td>1557</td>
<td>1623</td>
<td>783</td>
<td>762</td>
<td>855</td>
<td>614</td>
</tr>
</tbody>
</table>

<p><a id="table--Table summary enrichment"></a>
<div class="table-caption">
  <span class="table-number"><a href="#table--Table summary enrichment">Table 2</a></span>:
  Number of enriched annotation for each category and developmental stage.
</div></p>

<table>
<thead>
<tr>
<th>Enriched term</th>
<th>GW</th>
<th>GH</th>
<th>GB</th>
<th>YB</th>
<th>BB</th>
<th>DRY</th>
</tr>
</thead>

<tbody>
<tr>
<td>CDD</td>
<td>18</td>
<td>24</td>
<td>13</td>
<td>12</td>
<td>15</td>
<td>17</td>
</tr>

<tr>
<td>Cazy</td>
<td>2</td>
<td>4</td>
<td>1</td>
<td>6</td>
<td>2</td>
<td>0</td>
</tr>

<tr>
<td>GO</td>
<td>532</td>
<td>493</td>
<td>336</td>
<td>373</td>
<td>332</td>
<td>314</td>
</tr>

<tr>
<td>Gene3D</td>
<td>84</td>
<td>77</td>
<td>48</td>
<td>55</td>
<td>62</td>
<td>42</td>
</tr>

<tr>
<td>Hamap</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>1</td>
</tr>

<tr>
<td>IPR</td>
<td>320</td>
<td>326</td>
<td>222</td>
<td>229</td>
<td>241</td>
<td>183</td>
</tr>

<tr>
<td>KEGG</td>
<td>16</td>
<td>13</td>
<td>10</td>
<td>14</td>
<td>13</td>
<td>6</td>
</tr>

<tr>
<td>Merops</td>
<td>2</td>
<td>2</td>
<td>2</td>
<td>3</td>
<td>1</td>
<td>1</td>
</tr>

<tr>
<td>MetaCyc</td>
<td>11</td>
<td>15</td>
<td>9</td>
<td>18</td>
<td>0</td>
<td>0</td>
</tr>

<tr>
<td>MobiDBLite</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>

<tr>
<td>PANTHER</td>
<td>67</td>
<td>70</td>
<td>76</td>
<td>85</td>
<td>99</td>
<td>77</td>
</tr>

<tr>
<td>PIRSF</td>
<td>6</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>0</td>
</tr>

<tr>
<td>PRINTS</td>
<td>20</td>
<td>15</td>
<td>8</td>
<td>13</td>
<td>7</td>
<td>8</td>
</tr>

<tr>
<td>Pfam</td>
<td>116</td>
<td>118</td>
<td>71</td>
<td>81</td>
<td>91</td>
<td>57</td>
</tr>

<tr>
<td>ProDom</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>

<tr>
<td>ProSitePatterns</td>
<td>30</td>
<td>30</td>
<td>7</td>
<td>28</td>
<td>11</td>
<td>20</td>
</tr>

<tr>
<td>ProSiteProfiles</td>
<td>32</td>
<td>36</td>
<td>19</td>
<td>18</td>
<td>17</td>
<td>12</td>
</tr>

<tr>
<td>Reactome</td>
<td>38</td>
<td>28</td>
<td>18</td>
<td>12</td>
<td>22</td>
<td>21</td>
</tr>

<tr>
<td>SFLD</td>
<td>2</td>
<td>0</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
</tr>

<tr>
<td>SMART</td>
<td>42</td>
<td>44</td>
<td>25</td>
<td>21</td>
<td>28</td>
<td>24</td>
</tr>

<tr>
<td>SUPERFAMILY</td>
<td>86</td>
<td>81</td>
<td>55</td>
<td>52</td>
<td>63</td>
<td>40</td>
</tr>

<tr>
<td>TIGRFAM</td>
<td>3</td>
<td>0</td>
<td>2</td>
<td>2</td>
<td>2</td>
<td>1</td>
</tr>

<tr>
<td>eggNog</td>
<td>44</td>
<td>42</td>
<td>42</td>
<td>40</td>
<td>55</td>
<td>42</td>
</tr>

<tr>
<td>tcdb</td>
<td>13</td>
<td>13</td>
<td>12</td>
<td>12</td>
<td>13</td>
<td>10</td>
</tr>
</tbody>
</table>

<p>Now we would like to paint for each KEGG pathways the enzyme showing
up. To this aim we will use the <em>R</em>-package <strong><a href="http://bioconductor.org/packages/release/bioc/html/pathview.html" target="_blank">pathview</a></strong>, which <em>&rdquo;&hellip;maps and renders a wide variety of biological data on relevant pathway graphs. All users need is to supply their data and specify the target pathway. Pathview automatically downloads the pathway graph data, parses the data file, maps user data to the pathway, and render pathway graph with the mapped data. In addition, Pathview also seamlessly integrates with pathway and gene set (enrichment) analysis tools for large-scale and fully automated analysis&hellip;.&rdquo;</em>.
Since the annotation of the plant of interest might not be complete, the peptide mapping
process was achieved by also considering protein from other plants,
i.e. from more than one organism.</p>

<p>The problem is that pathview either works with a species specific set
of genes ID or with the corresponding KEGGID. A solution is to convert
the gene ID to KEGG ID. To this aim we generated a gene ID to EC
dictionary from the <strong>IPR</strong> annotation. The EC to KEGG was obtained the
<a href="http://rest.kegg.jp" target="_blank">KEGG Rest Api</a>:</p>

<pre><code class="language-bash">  curl -f http://rest.kegg.jp/link/ec/ko | sed -r 's/ko://' | sed -r 's/ec:/EC:/' &gt; KO2EC.list
#KO2EC file looks then like
 #KEGG    EC
 #K00001  EC:1.1.1.1
 #K00002  EC:1.1.1.2
 #K00003  EC:1.1.1.3
 #K00004  EC:1.1.1.4
 #K00004  EC:1.1.1.303
 #K00005  EC:1.1.1.6
 #K00006  EC:1.1.1.8
 #K00007  EC:1.1.1.11
 #K00008  EC:1.1.1.14
</code></pre>

<div class="src-block-caption">
  <span class="src-block-number">Code Snippet 1</span>:
  Downloading KO / EC file
</div>

<p>Finally the enrichment files have the following formatting</p>

<pre><code class="language-text">&quot;KEGGID&quot;        &quot;Pvalue&quot;        &quot;OddsRatio&quot;     &quot;ExpCount&quot;      &quot;Count&quot; &quot;Size&quot;  &quot;Term&quot;  &quot;fdr&quot;   &quot;desc&quot;  &quot;genes&quot;
 &quot;ko00630&quot;       1.48869690281379e-05    8.83636363636364        1.69491525423729        9       25      NA      0.000699687544322481    &quot;Glyoxylate_and_dicarboxylate_metabolism&quot;       &quot;A0A067JFR1, A0A067JMA6, A0A067K0M8, A0A067KN15, A5X4N8, C6F1E5, K9MCB1, Q84XZ9, W9SPU5&quot;
 &quot;ko00020&quot;       3.35093741041604e-05    11.8849902534113        1.08474576271186        7       16      NA      0.00078747029144777     &quot;Citrate_cycle_(TCA_cycle)&quot;     &quot;A0A067JFR1, A0A067JMA6, A0A067JUK1, A0A067K488, A0A067LBG3, A0A0B2SEY3, I1MGE6&quot;
 &quot;ko00720&quot;       0.000359810652978868    7.08187134502924        1.49152542372881        7       22      NA      0.00485214527417342     &quot;Carbon_fixation_pathways_in_prokaryotes&quot;       &quot;A0A067JFR1, A0A067JMA6, A0A067JUK1, A0A067LBG3, A0A0B2SEY3, I1MGE6, J9Y1K3&quot;
 &quot;ko00380&quot;       0.000412948533972206    12.3446327683616        0.745762711864407       5       11      NA      0.00485214527417342     &quot;Tryptophan_metabolism&quot; &quot;A0A067K488, C6F1E5, K9MCB1, Q84XZ9, W9SPU5&quot;
</code></pre>

<h2 id="processing">PROCESSING</h2>

<p>Once we have the set of enriched KEGG pathway with the corresponding
gene name, the conversion files from GeneID-&gt;EC-&gt;KEGGID, we
can start genering colorised KEGG pathways.</p>

<pre><code class="language-R">#In work2/fatemeh/proteinFunctionalAnnotation/enrichment2
#Library used to colorise KEGG pathways
library(pathwview)

#Read in the GENE-&gt;EC dictionary
dictUniEC&lt;-read.table(&quot;./helperfile/Gene_EC.dictionary&quot;,header=T)
#Read the EC2KO dictionary
dictECKEGGID&lt;-read.table(&quot;/media/htafer/work2/share/database/KO2EC.list&quot;,header=T)
#Replace EC: with &quot;&quot;
dictECKEGGID$EC&lt;-gsub(&quot;EC:&quot;,&quot;&quot;,dictECKEGGID$EC)
#Get the list of files containing enriched KEGG
files&lt;-list.files(path=&quot;.&quot;,pattern=&quot;list.KEGG.csv$&quot;)
#for each file
for (f in files){
     # get the data
     data&lt;-read.table(f,quote='&quot;',header=T)
     # select enrichment q-balue &lt; 0.05
     data&lt;-data[data$fdr&lt;0.05,]
     # replace ko with &quot;&quot;
     data$KEGGID&lt;-gsub(&quot;ko&quot;,&quot;&quot;,data$KEGGID)
     # replace ec with EC
     data$ECs&lt;-gsub(&quot;ec:&quot;,&quot;EC:&quot;,data$genes)
     # for each gene inside the enriched annotation
     for(i in 1:dim(data)[1]){
	#Get contig for give kegg
	contigs&lt;-strsplit(as.character(data$ECs),&quot;,&quot;)[[i]]
	#Rmove white space
	contigs&lt;-gsub(&quot; &quot;,&quot;&quot;,contigs)
	#id conversion
	temp&lt;-unique(dictECKEGGID$KEGG[dictECKEGGID$EC %in% unique(dictUniEC$ec[dictUniEC$uni %in% contigs])])

					 #                                        #Read dictionary
					 #
					 #    matchLOC&lt;-dictLOCEC[dictLOCEC$EC %in% contigs,1]
					 #                                        #Remove the loc part since it is not used in KEGG
					 #    contigs&lt;-gsub(&quot;LOC&quot;,&quot;&quot;,matchLOC)
					 #                                        #Start analysis
					 #    pathview(contig,pathway.id=sprintf(&quot;%05s&quot;,data$KEGGID[i]),species=&quot;jcu&quot;,out.suffix=paste(data$KEGGID[i],&quot;_&quot;,gsub(&quot; &quot;,&quot;&quot;,data$ECs[i]),sep=&quot;&quot;))
	#Plot with pathview
	 try(pathview(as.character(temp),pathway.id=data$KEGGID[i],species=&quot;ko&quot;,gene.idtype=&quot;kegg&quot;,plot.col.key= FALSE,out.suffix=paste(data$KEGGID[i],f,sep=&quot;_&quot;)))
     }
 }
</code></pre>

<div class="src-block-caption">
  <span class="src-block-number">Code Snippet 2</span>:
  <i>R</i>-code used to colorize the overrpresented KEGG pathways with the correspponding enzymes
</div>

<h2 id="results">RESULTS</h2>

<p>The resulting pathways are now showing in red the genes with at least one peptide.</p>

<p><a id="org9655a82"></a></p>




<figure>

<img src="/img/ko00020.00020_jatCur.yb.csv.list.KEGG.csv.png" alt="Figure 1: Colorized KEGG pathway rendered by pathview" />



<figcaption data-pre="Figure " data-post=":" >
  
  <p>
    Figure 1: Colorized KEGG pathway rendered by pathview
    
    
    
  </p> 
</figcaption>

</figure>

<h2 id="conclusion">Conclusion</h2>

<p>Pathview is a great tool to annotate metabolic pathway from KEGG. The
main difficulty is to get the right gene ID for pathview to produce
the desired plot. Once this is ensured publication ready plots are
easily produced by this great package.</p>

    </div>

    


<div class="article-tags">
  
  <a class="badge badge-light" href="/tags/proteomics/">proteomics</a>
  
  <a class="badge badge-light" href="/tags/kegg/">KEGG</a>
  
  <a class="badge badge-light" href="/tags/genes/">genes</a>
  
  <a class="badge badge-light" href="/tags/plant/">plant</a>
  
  <a class="badge badge-light" href="/tags/enrichment/">enrichment</a>
  
  <a class="badge badge-light" href="/tags/pathview/">pathview</a>
  
  <a class="badge badge-light" href="/tags/plot/">plot</a>
  
  <a class="badge badge-light" href="/tags/visualization/">visualization</a>
  
  <a class="badge badge-light" href="/tags/rest/">REST</a>
  
  <a class="badge badge-light" href="/tags/api/">api</a>
  
</div>




    
      






  







<div class="media author-card" itemscope itemtype="http://schema.org/Person">
  
  
  <img class="portrait mr-3" src="/author/admin/avatar_hu52a603635ecebd45650b162dadabb4e5_12861_250x250_fill_q75_box_center.jpg" itemprop="image" alt="Avatar">
  

  <div class="media-body">
    <h5 class="card-title" itemprop="name"><a href="/authors/admin">Erco 1976</a></h5>
    <h6 class="card-subtitle">Senior Scientist Bioinformatics</h6>
    
    <ul class="network-icon" aria-hidden="true">
      
      
      
      
        
      
      
      
      
      
        
      
      <li>
        <a itemprop="sameAs" href="https://www.researchgate.net/profile/Hakim_Tafer" target="_blank" rel="noopener">
          <i class="fab fa-researchgate"></i>
        </a>
      </li>
      
      
      
      
      
      
      
      
        
      
      <li>
        <a itemprop="sameAs" href="https://scholar.google.co.uk/citations?user=86NfI6IAAAAJ" target="_blank" rel="noopener">
          <i class="ai ai-google-scholar"></i>
        </a>
      </li>
      
      
      
      
        
      
      
      
      
      
        
      
      <li>
        <a itemprop="sameAs" href="https://github.com/htafer" target="_blank" rel="noopener">
          <i class="fab fa-github"></i>
        </a>
      </li>
      
    </ul>
  </div>
</div>



      
      
      <div class="article-widget">
        <div class="hr-light"></div>
        <h3>Related</h3>
        <ul>
          
          <li><a href="/posts/kegg2/">KEGG analysis of regulated genes</a></li>
          
        </ul>
      </div>
      
    

    

    
<section id="comments">
  <div id="disqus_thread"></div>
<script type="application/javascript">
    var disqus_config = function () {
    
    
    
    };
    (function() {
        if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
            document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
            return;
        }
        var d = document, s = d.createElement('script'); s.async = true;
        s.src = '//' + "Erco1976" + '.disqus.com/embed.js';
        s.setAttribute('data-timestamp', +new Date());
        (d.head || d.body).appendChild(s);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
</section>



  </div>
</article>

<div class="container">
  <footer class="site-footer">
  

  <p class="powered-by">
    &copy; 2018-2019 &middot; 

    Powered by the
    <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
    <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

    
    <span class="float-right" aria-hidden="true">
      <a href="#" id="back_to_top">
        <span class="button_icon">
          <i class="fas fa-chevron-up fa-2x"></i>
        </span>
      </a>
    </span>
    
  </p>
</footer>

</div>


<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title">Cite</h5>
        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body">
        <pre><code class="tex hljs"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-outline-primary my-1 js-copy-cite" href="#" target="_blank">
          <i class="fas fa-copy"></i> Copy
        </a>
        <a class="btn btn-outline-primary my-1 js-download-cite" href="#" target="_blank">
          <i class="fas fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

    

    
    

    
    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha512-+NqPlbbtM1QqiK8ZAo4Yrj2c4lNQoGv8P79DPtKzj++l5jnN39rHA/xsqn8zE9l0uSoxaCdrOgFs6yjyfbBxSg==" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.3/imagesloaded.pkgd.min.js" integrity="sha512-umsR78NN0D23AzgoZ11K7raBD+R6hqKojyBZs1w8WvYlsI+QuKRGBx3LFCwhatzBunCjDuJpDHwxD13sLMbpRA==" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.1.3/js/bootstrap.min.js" integrity="sha256-VsEqElsCHSGmnmHXGQzvoWjWwoznFSZc6hs7ARLRacQ=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.4/isotope.pkgd.min.js" integrity="sha512-VDBOIlDbuC4VWxGJNmuFRQ0Li0SKkDpmGyuhAG5LTDLd/dJ/S0WMVxriR2Y+CyPL5gzjpN4f/6iqWVBJlht0tQ==" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.js" integrity="sha256-X5PoE3KU5l+JcX+w09p/wHl9AzK333C4hJ2I9S5mD4M=" crossorigin="anonymous"></script>

      
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js" integrity="sha256-/BfiIkHlHoVihZdc6TFuj7MmJ0TWcWsMXkeDFwhi0zw=" crossorigin="anonymous"></script>
        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/r.min.js"></script>
        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/lisp.min.js"></script>
        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/bash.min.js"></script>
        
        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/C.min.js"></script>
        
      

      
      
    

    
    

    
    
    
    <script id="dsq-count-scr" src="//Erco1976.disqus.com/count.js" async></script>
    

    
    
    <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script>
      const search_index_filename = "/index.json";
      const i18n = {
        'placeholder': "Search...",
        'results': "results found",
        'no_results': "No results found"
      };
      const content_type = {
        'post': "Posts",
        'project': "Projects",
        'publication' : "Publications",
        'talk' : "Talks"
        };
    </script>
    

    
    

    
    
    <script id="search-hit-fuse-template" type="text/x-template">
      <div class="search-hit" id="summary-{{key}}">
      <div class="search-hit-content">
        <div class="search-hit-name">
          <a href="{{relpermalink}}">{{title}}</a>
          <div class="article-metadata search-hit-type">{{type}}</div>
          <p class="search-hit-description">{{snippet}}</p>
        </div>
      </div>
      </div>
    </script>
    

    
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/3.2.1/fuse.min.js" integrity="sha256-VzgmKYmhsGNNN4Ph1kMW+BjoYJM2jV5i4IlFoeZA9XI=" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>
    

    
    

    
    
    
    
    
    
    
      
    
    
    
    
    <script src="/js/academic.min.7903dc64949b0cecbfa70e940e162fa7.js"></script>

  </body>
</html>

