+++
title = "RNA in phylogenetic reconstruction"
date = "2014-12-31"
draft = false

# Authors. Comma separated list, e.g. `["Bob Smith", "David Jones"]`.
authors = ["Roman R. Stocsits", "Harald Letsch", "Karen Meusemann", "Bjoern M. Reumont", "Bernhard Misof", "Jana Hertel", "Hakim Tafer", "Peter F. Stadler"]

# Publication type.
# Legend:
# 0 = Uncategorized
# 1 = Conference paper
# 2 = Journal article
# 3 = Manuscript
# 4 = Report
# 5 = Book
# 6 = Book section
publication_types = ["1"]

# Publication name and optional abbreviated version.
# publication = "In *International Conference on Multimedia and Expo Workshops (ICMEW)*, IEEE."
# publication_short = "In *ICMEW*"

# Abstract and optional shortened version.
abstract = " Ribosomal RNA (rRNA) genes are the most widely used data source in phylogenetic reconstruction. They are highly structured, with large parts of the mol- ecules exhibiting very strong conservation of their base pairing patterns. Since also sequence conservation varies dramatically between different regions of rRNA genes, these data are informative on a wide range of phylogenetic timescales, ranging from recent to ancient splits. But individual columns of rRNA alignments are not inde- pendent. This is a consequence of biologically functional secondary structures with highly varying degrees of conservation. Therefore, any correlations within sequence alignments of structurally functional RNA will distort the phylogenetic signal and/or lead to gross overestimates of tree stability. On the other hand, alignment accuracy can be improved substantially by incorporating secondary structure conservation. Maximum Likelihood and Bayesian approaches are amenable to using RNA-specific substitution models that treat conserved base pairs appropriately, but they require accurate secondary structure models as input. Structure prediction algorithms for single RNA sequences are well known and widely used; but it is not straightforward to apply these predictions for phylogenetic purposes. The main limitation is that accuracy of thermodynamic folding algorithms declines sharply as the length of the RNA increases, this is in part due to inaccuracies within the thermodynamic folding parameters, in part caused by the kinetics of folding process and tertiary interac- tions, and even more in the fact that RNA and protein components of the ribosome are tightly packed and thus mutually influence their folds. The functional rRNA struc- tures, therefore, cannot reasonably be expected to be identical with the minimum free energy structures of isolated rRNAs computed by thermodynamic folding algorithms. Our RNAsalsa approach has been the first accurate and easy-to-use tool to address those implications mentioned. It is available for computing structure-aware align- ments and consensus structures and can deal with large sets of large rRNAs. Multiple pairwise sequence alignments, intersections with individualized structural constraint modeling, and computationally derived collections of optimized subgroup consen- sus structures allow to extract a refined individualized structure constraint for each single sequence by means of a majority voting procedure. That way an extensively optimized set of individual constraints serves in minimum energy folding to obtain the final structural annotation for each input sequence. Further, the structure-anno- tated input sequences are aligned by a structure-aware alignment method, resulting in a final alignment and a corresponding consensus structure which are ready for phylogenetic analysis based on RNA-specific substitution models. The difference to DNA substitution models is that structure-aware models consider pairs of nucleotides as their elementary states rather than single nucleotides. Further, studies suggest that RNA secondary structures are by themselves also phylogenetically informative since structural variations are specific to individual clades. This has been reported for a wide variety of non-coding RNAs, including rRNAs, 7SK RNAs, and mitochondrial tRNAs. But in all cases the massive epistasis needs to be taken into account explicitly which is introduced by strong selection on secondary structure features."
# abstract_short = "A mobile visual clothing search system is presented whereby a smart phone user can either choose a social networking image or capture a new photo of a person wearing clothing of interest and search for similar clothing in a large cloud-based ecommerce database. The phone's GPS location is used to re-rank results by retail store location, to inform the user of local stores where similar clothing items can be tried on."

# Is this a selected publication? (true/false)
selected = false

# Projects (optional).
#   Associate this publication with one or more of your projects.
#   Simply enter your project's folder or file name without extension.
#   E.g. `projects = ["deep-learning"]` references 
#   `content/project/deep-learning/index.md`.
#   Otherwise, set `projects = []`.
projects = []

# Tags (optional).
#   Set `tags = []` for no tags, or use the form `tags = ["A Tag", "Another Tag"]` for one or more tags.
tags = []

# Links (optional).
# url_pdf = "10.1515/9783110277524.531"
# url_preprint = "#"
# url_code = "#"
# url_dataset = "#"
# url_project = ""
# url_slides = "#"
# url_video = "#"
# url_poster = "#"
# url_source = "#"

# Custom links (optional).
#   Uncomment line below to enable. For multiple links, use the form `[{...}, {...}, {...}]`.
# url_custom = [{name = "Custom Link", url = "http://example.org"}]

# Digital Object Identifier (DOI)
doi = "10.1515/9783110277524.531"

# Does this page contain LaTeX math? (true/false)
math = false

# Featured image
# To use, add an image named `featured.jpg/png` to your page's folder. 
# [image]
#  # Caption (optional)
#  caption = "Image credit: [**Unsplash**](https://unsplash.com/photos/pLCdAaMFLTE)"
#
#  # Focal point (optional)
#  # Options: Smart, Center, TopLeft, Top, TopRight, Left, Right, BottomLeft, Bottom, BottomRight
#  focal_point = ""
+++

